### ISSUE TITLE
Version 1.0 Hit List

### ISSUE THREAD

**Issue by sdwilsh** [sdwilsh/siobrultech-protocols#146]
This exists to track the work we need to do in order to set the version to 1.0
- [ ] Update `Development Status :: 4 - Beta` classifier in `setup.cfg`
- [ ] Fix import paths for `gem` for things that are not `gem`-dependent

**Comment by sdwilsh** [comment]
@jkeljo, I think there was a thing or two you mentioned that I'm drawing a blank on here as well (and I know we talked about it, but I also can't find that conversation).

**Comment by jkeljo** [comment]
Yeah, added. Convo was in https://github.com/sdwilsh/siobrultech-protocols/pull/127

**Comment by jkeljo** [comment]
Ok so with #205 landed we need to either decide that it's OK for that small breaking change to happen in 0.13, or do these things and cut 1.0. Thoughts?

If you think we should go for 1.0 now, do you have any thoughts on the right code organization? At a quick glance I see four categories of code in `siobrultech_protocols.gem` today:
1 - low-level stuff that is used by both GEM and ECM code, e.g. `fields.py`, the `Packet` and `PacketFormat` classes
2 & 3 - GEM or ECM-specific stuff, like the packet format definitions in `packets.py` or the API configurations in `api.py`; all the stuff in `const.py` looks GEM-specific
4 - the high-level interface to the library, which can handle both GEM and ECM (`protocol.py`, most of `api.py`

One way to do this might be:
- Everything in 1 & 4 moves to siobrultech_protocols
- 2 moves to siobrultech_protocols/gem.py
- 3 moves to siobrultech_protocols/ecm.py

wdyt?

**Comment by jkeljo** [comment]
I suppose another question is do we want to keep the `siobrultech_protocols` base package, or should we move both this and `greeneye_monitor` to using namespace packages and root them at a common base like `brultech`? That might be too much bikeshedding.

**Comment by sdwilsh** [comment]
> Ok so with https://github.com/sdwilsh/siobrultech-protocols/pull/205 landed we need to either decide that it's OK for that small breaking change to happen in 0.13, or do these things and cut 1.0. Thoughts?

Technically, we're pre-1.0, so I think breaking changes are okay.  I think you and I are the [only ones using this right now](https://libraries.io/pypi/siobrultech-protocols), anyway!

> One way to do this might be:
> 
>     Everything in 1 & 4 moves to siobrultech_protocols
>     2 moves to siobrultech_protocols/gem.py
>     3 moves to siobrultech_protocols/ecm.py
> 
> wdyt?

This is what I had in mind as well!

> I suppose another question is do we want to keep the `siobrultech_protocols` base package, or should we move both this and `greeneye_monitor` to using namespace packages and root them at a common base like `brultech`? That might be too much bikeshedding.

Oh interesting.  I think it might be nice for us to merge all this, but we could also do that post 1.0.  It might even be nice to consolidate some repositories so we can share some common things as well.

**Comment by jkeljo** [comment]
Ok, I'll cut 0.13 so I can get that bugfix into HACS, and we can work on the code organization for 1.0.
